### PR TITLE
feat: add wallet distribution chart

### PR DESCRIPTION
**Pull request by SilVdB**
## Summary
- Add wallet distribution chart to visualize token holder concentration
- Implement ORPC API endpoint with dynamic percentage-based buckets
- Create interactive bar chart component for token details page

### Backend
- New ORPC endpoint `/token/stats/{tokenAddress}/wallet-distribution`
- Dynamic bucket calculation (0-2%, 2-10%, 10-20%, 20-40%, 40-100% of max balance)
- Comprehensive test coverage for schema validation and edge cases

### Frontend  
- `AssetWalletDistributionChart` component using existing `BarChartComponent`
- Integration into token details page with 3-column grid layout
- Translation support for chart title, description, and labels

## Testing
- ✅ All unit tests passing (25 tests for wallet distribution)
- ✅ Linting and type checking pass
- ✅ Manual testing with various token distributions

## Screenshots
\Wallet Distribution Chart Design

## Related
- Linear Issue: [ENG-3554](https://linear.app/settlemint/issue/ENG-3554/add-wallet-distribution-chart)

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: kit/dapp/src/orpc/routes/token/routes/stats/wallet-distribution.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+import { theGraphMiddleware } from "@/orpc/middlewares/services/the-graph.middleware";
+import { tokenRouter } from "@/orpc/procedures/token.router";
+import { z } from "zod";
+
+/**
+ * GraphQL query to fetch token balances for wallet distribution calculation
+ * Retrieves all non-zero balances for a specific token
+ */
+const TOKEN_BALANCES_QUERY = theGraphGraphql(`
+  query TokenBalancesForDistribution($tokenId: ID!) {
```
Comment (janb87) [inline kit/dapp/src/orpc/routes/token/routes/stats/wallet-distribution.ts:11]: Use the optimised data from the subgraph, see https://github.com/settlemint/asset-tokenization-kit/blob/main/kit/subgraph/test/token-distribution-stats.spec.ts

**Review by janb87 — CHANGES_REQUESTED** [review]
Subgraph already indexes this

File: kit/dapp/src/orpc/routes/token/routes/stats/wallet-distribution.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    token(id: $tokenId) {
+      balances(where: { value_gt: "0" }) {
```
[the remark this answers is not in the document]
Comment (SilVdB) [reply kit/dapp/src/orpc/routes/token/routes/stats/wallet-distribution.ts:13]: For this specific query we don't need FetchAll because:

**Review by janb87 — APPROVED** [review]
